app/assets/javascripts/redux/modules/models/listenSignals.js

```diff
@@ -140,6 +141,16 @@ export const reducer = handleActions({
 
   [LISTEN_SIGNALS_PUT_REQUEST_FAIL]: handlesListenSignalFailResponse,
 
+  [PROMOTION_SIGNAL_POST_REQUEST_SUCCESS]: (state, action) => {
+    const promotionalTweet = _.get(action.payload, 'promotional_tweet', {});
+    const id = promotionalTweet.id;
+    const signalId = promotionalTweet.listen_signal_id
```

Comment:
You don't need to do anything here with the `signalId`. You _should_ be listening to this action within the `listenSignals`reducer to add the promotional tweet id to the array of promotional tweets for the listen signal.